src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/ProbabilisticMap.cs

```diff
@@ -56,23 +70,114 @@ public ProbabilisticMap(ReadOnlySpan<char> values)
-                charMap |= 1u;
+                SetCharBit(ref charMap, 0);
             }
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static void SetCharBit(ref uint charMap, byte value) =>
-            Unsafe.Add(ref charMap, (uint)value & IndexMask) |= 1u << (value >> IndexShift);
+        private static void SetCharBit(ref uint charMap, byte value)
+        {
+            if (IsVectorizationSupported)
+            {
+                Unsafe.Add(ref Unsafe.As<uint, byte>(ref charMap), value & VectorizedIndexMask) |= (byte)(1u << (value >> VectorizedIndexShift));
+            }
+            else
+            {
+                Unsafe.Add(ref charMap, value & PortableIndexMask) |= 1u << (value >> PortableIndexShift);
+            }
+        }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool IsCharBitSet(ref uint charMap, byte value) =>
-            (Unsafe.Add(ref charMap, (uint)value & IndexMask) & (1u << (value >> IndexShift))) != 0;
+        private static bool IsCharBitSet(ref uint charMap, byte value) => IsVectorizationSupported
+            ? (Unsafe.Add(ref Unsafe.As<uint, byte>(ref charMap), value & VectorizedIndexMask) & (1u << (value >> VectorizedIndexShift))) != 0
+            : (Unsafe.Add(ref charMap, value & PortableIndexMask) & (1u << (value >> PortableIndexShift))) != 0;
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static bool Contains(ref uint charMap, ReadOnlySpan<char> values, int ch) =>
             IsCharBitSet(ref charMap, (byte)ch) &&
             IsCharBitSet(ref charMap, (byte)(ch >> 8)) &&
-            values.Contains((char)ch);
+            Contains(values, (char)ch);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool Contains(ReadOnlySpan<char> values, char ch) =>
+            SpanHelpers.NonPackedContainsValueType(
+                ref Unsafe.As<char, short>(ref MemoryMarshal.GetReference(values)),
+                (short)ch,
+                values.Length);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector256<byte> ContainsMask32CharsAvx2(Vector256<byte> charMapLower, Vector256<byte> charMapUpper, ref char searchSpace)
+        {
+            Vector256<ushort> source0 = Vector256.LoadUnsafe(ref searchSpace);
+            Vector256<ushort> source1 = Vector256.LoadUnsafe(ref searchSpace, (nuint)Vector256<ushort>.Count);
+
+            Vector256<byte> sourceLower = Avx2.PackUnsignedSaturate(
+                (source0 & Vector256.Create((ushort)255)).AsInt16(),
+                (source1 & Vector256.Create((ushort)255)).AsInt16());
+
+            Vector256<byte> sourceUpper = Avx2.PackUnsignedSaturate(
+                Vector256.ShiftRightLogical(source0, 8).AsInt16(),
+                Vector256.ShiftRightLogical(source1, 8).AsInt16());
```

Comment:
You could use the operators here:
```suggestion
                (source0 >>> 8).AsInt16(),
                (source1 >>> 8).AsInt16());
```